Formula/g/guichan.rb

```diff
@@ -0,0 +1,168 @@
+                        int dstY,
+                        int width,
+                        int height)
+          {
+              if (const MyImage* myImage = dynamic_cast<const MyImage*>(image))
+              {
+                  const gcn::ClipRectangle &clipArea = getCurrentClipArea();
+                  std::cout << "Rendering image " << myImage->getPath() << " at ("
+                            << clipArea.xOffset + dstX << ", " << clipArea.yOffset + dstY << ")"
+                            << std::endl;
+              }
+          }
+
+          void drawPoint(int x, int y) {};
+          void drawLine(int x1, int y1, int x2, int y2) {};
+          void drawRectangle(const gcn::Rectangle& rectangle) {};
+          void fillRectangle(const gcn::Rectangle& rectangle) {};
+          void setColor(const gcn::Color& color) { mColor = color; };
+          const gcn::Color& getColor() const { return mColor; };
+
+      private:
+          gcn::Color mColor;
+      };
+
+      class MyImageLoader : public gcn::ImageLoader
+      {
+      public:
+          MyImageLoader() {}
+
+          gcn::Image* load(const std::string& filename, bool convertToDisplayFormat = true)
+          {
+              gcn::Image* image = new MyImage(filename);
+              if (convertToDisplayFormat)
+                  image->convertToDisplayFormat();
+              return image;
+          }
+      };
+
+      class MyFont : public gcn::Font
+      {
+      public:
+          int getWidth(const std::string &text) const { return 0; }
+          int getHeight() const { return 0; }
+
+          void drawString(gcn::Graphics *graphics, const std::string &text, int x, int y)
+          {
+              const gcn::ClipRectangle &clipArea = graphics->getCurrentClipArea();
+              std::cout << "Drawing string: " << text << " at (" << clipArea.xOffset + x << ", "
+                        << clipArea.yOffset + y << ")" << std::endl;
+          }
+      };
+
+      int main(int argc, char **argv)
+      {
+          MyInput myInput;
+          MyGraphics myGraphics;
+          MyImageLoader myImageLoader;
+          MyFont myFont;
+
+          gcn::Image::setImageLoader(&myImageLoader);
+          gcn::Widget::setGlobalFont(&myFont);
+
+          gcn::Gui gui;
+          gui.setInput(&myInput);
+          gui.setGraphics(&myGraphics);
+
+          gcn::Icon icon("icon.png");
+          gcn::Label label("Hello, Guichan!");
+          gcn::Container container;
+          container.setDimension(gcn::Rectangle(0, 0, 200, 200));
+          container.add(&label, 10, 10);
+          container.add(&icon, 10, 30);
+          gui.setTop(&container);
+
+          gui.logic();
+          gui.draw();
+
+          return 0;
+      }
+    EOS
+
+    flags = [
+      "-L#{lib}", "-lguichan"
+    ]
+
+    flags << (OS.mac? ? "-lc++" : "-lstdc++")
+
+    system ENV.cc, "helloworld.cpp", ENV.cppflags,
+                   *flags, "-o", "helloworld"
+    system "./helloworld"
```

Comment:
maybe better to use pkgconf to set the flags?

Reply:
I think this would work
```suggestion
    pkg_config_flags = shell_output("pkg-config --cflags --libs guichan").chomp.split
    system ENV.cxx, "helloworld.cpp", *pkg_config_flags, "-o", "helloworld"
    system "./helloworld"
```

Reply:
Ah, I thought it was about the `(OS.mac? ? "-lc++" : "-lstdc++")` part, but maybe it is indeed superfluous.

Indeed using pkg-config for linking the test should be better since it verifies whether that part works as well.

Reply:
Ah, `pkg-config` can't be found and I don't think I want to introduce a dependency on it, so I've reverted this change again.

Reply:
You can add it as

```
depends_on "pkgconf" => :test
```

That is test only dependency.

Reply:
No, it's best to add a `depends_on "pkg-config" => :test`

Reply:
So this would make only the test depend on pkg-config? I will do a PR for that then, thanks!

Reply:
@SMillerDev I guess actually the version suggested by @daeho-ro is preferred, since I got this during `brew audit guichan`:
```
guichan
  * Dependency 'pkg-config' is an alias; use the canonical name 'pkgconf'.
Error: 1 problem in 1 formula detected.
```
Apparently they're using [pkgconf](https://formulae.brew.sh/formula/pkgconf) to provide [pkg-config](https://www.freedesktop.org/wiki/Software/pkg-config/) in Homebrew.